### ISSUE TITLE
Prompt ignores chosen container

### ISSUE THREAD

**Issue by FulcrumA** [CleverRaven/Cataclysm-DDA#23735]
Game version: Currently 0.C-27803

Operating system: Windows 7

Tiles or curses: Tiles

Mods active: More Survival Tools, No Explosive Zombies, No Fungal Monsters, StatsThroughSkill, Vehicle Additions Pack, Tanks and Other Vehicles, Bionics Systems Mod, Boats, Icecoon's Arsenal, Filthy Clothing, Makeshift Items Mod, More Buildings, No Flaming Weapons, Disable Mutagens, Disable NPC Needs, No Antique Firearms, No Fictional Guns, NPC Traits

### Expected behavior

In this situation:

![uhoh](https://user-images.githubusercontent.com/38487938/40044221-0b870b1c-5827-11e8-8c39-6ab43c1a0fe3.png)

Choosing a container to refill the jug refills the jug with content of the container to the extent of jug's volume.

### Actual behavior
No matter the choice, first source container is chosen - in the situation from the screen above, jug gets refilled with clean water from the canteen even when it's unpurified water tanks that are chosen.

### Steps to reproduce the behavior
Acquire several similar containers containing liquids, get another, empty container, attempt to refill the empty container from the previous, filled ones.

**Comment by DaviBones** [comment]
I have seen this bug many times, it has done really frustrating things such as load sealed perishable liquids (soups, red sauce, etc) into my waterskin when I was just trying to get some water.

Like most of the bugs I have encountered with the new liquid-reloading feature, I have found this bug sporadic and difficult to nail down or reproduce on demand, though I can confirm it is exclusive to empty containers as was mentioned.

**Comment by HAnthonyHoyt** [comment]
Okay, I spent some time trying to replicate this and here's something I ran into.  Not sure if it's directly related but here's what I found.

There's a bug with reloading using items found on the ground.  In this case, I did the following
1. Have an empty 'gallon jug' in my inventory
1. Placed a 'tin can of red sauce' on the ground
1. Attempt to 'reload' the jug with the red sauce

This causes the application to crash on windows.  Note, this is with the latest commit as of 5/17/2018 10pm ET on Windows 10 built using VS 2017 Community edition.

Callstack
>	Cataclysm.exe!cata::optional<islot_gun>::has_value() Line 119	C++
 	Cataclysm.exe!item::is_gun() Line 3558	C++
 	Cataclysm.exe!item::has_flag(const std::basic_string<char,std::char_traits<char>,std::allocator<char> > & f) Line 2744	C++
 	Cataclysm.exe!activity_handlers::reload_finish(player_activity * act, player * p) Line 1284	C++
 	[External Code]	
 	Cataclysm.exe!activity_type::call_finish(player_activity * act, player * p) Line 94	C++
 	Cataclysm.exe!player_activity::do_turn(player & p) Line 122	C++
 	Cataclysm.exe!game::process_activity() Line 1613	C++
 	Cataclysm.exe!game::do_turn() Line 1453	C++
 	Cataclysm.exe!SDL_main(int argc, char * * argv) Line 523	C++
 	Cataclysm.exe!main_utf8(int argc, char * * argv) Line 127	C
 	Cataclysm.exe!WinMain(HINSTANCE__ * hInst, HINSTANCE__ * hPrev, char * szCmdLine, int sw) Line 191	C

Exception:
> Unhandled exception thrown: read access violation.
> **this** was 0xFFFFFFFFFFFFFE97. occurred

Anyone else can verify this?

**Comment by Night-Pryanik** [comment]
@HAnthonyHoyt cannot confirm on 0.C- 7402, Windows XP SP3 x86 curses.

**Comment by HAnthonyHoyt** [comment]
I'm able to repeat the issue on Windows 10, using the curses build of 0.C-27848-g4bcbac0.  Here are my steps

1. Create a new character
1. In the debug menu, create the following
    1. MOLLE pack
    1. tin can of meat soup
    1. empty gallon jug
1. Wear MOLLE pack
1. Drop tin can on floor
1. Refill gallon jug with soup on floor

I got this error:
![image](https://user-images.githubusercontent.com/23622706/40217360-5d851f3e-5a3a-11e8-896e-3eff02fb9b8f.png)

I was also able to get the game to crash with the Windows Tiles build as well.  So it's not just my local build.

**Comment by Night-Pryanik** [comment]
Hmm, strange. Still can't reproduce.

![default](https://user-images.githubusercontent.com/11132525/40232069-081b0d62-5aa6-11e8-818e-75ab0b35cb5f.png)

**Comment by HAnthonyHoyt** [comment]
A small update on the reloading items from floor issue.  I've managed to test the issue on several platforms and here's what I've found

Issue is reproducible on:
* Windows 10 x64 Tiles
* Windows 10 x64 Curses
* MacOS 10.13 Tiles
* MacOS 10.13 Curses

Not reproducible on:
* Ubuntu 10.04 Tiles
* Ubuntu 10.04 Curses
* Windows 7 Tiles

So it's platform specific.  At the moment, I'm not sure what's the difference between the various platforms.
